lib/include/tkge/assetLoader.hpp

```diff
@@ -18,7 +18,12 @@ namespace tkge
 
 			for (const auto& path : paths)
 			{
-				if (std::filesystem::exists(path / fileName)) { return std::make_unique<T>((path / fileName).string()); }
+				if (std::filesystem::exists(path / fileName))
+				{
+					auto asset = std::make_unique<T>();
+					asset->Load(Assets::ReadonlyByteStream{(path / fileName).string()});
+					return asset;
+				}
 			}
```

Comment:
File system / reading logic needs to be decoupled from asset systems in the next PR. Allocating a vector of paths for each asset load is not needed, the file loader can cache those paths once when the "mount point" (assets root dir) is set. IRL there will only be one base path, so it doesn't even need to be a vector as such.